utils.go

```diff
@@ -86,6 +88,12 @@ func formatError(err error) error {
 		return err
 	}
 
+	if pe, ok := err.(*fs.PathError); ok {
+		if errno, ok := pe.Err.(syscall.Errno); ok && errno == syscall.EISDIR {
```

Comment:
`syscall` may not work on `darwin` and `windows`.

Reply:
I'm not sure how to handle `PathError` here. Can I use the `is a directory` error message as the condition?

Reply:
I don't think use `is a directory` is a good idea here.

How about:

```go
	if pe, ok := err.(*os.PathError); ok {
		switch pe.Err {
		case syscall.EISDIR:
			return fmt.Errorf("%w: %v", services.ErrObjectModeInvalid, err)
		}
	}
```